app/controllers/keywords_controller.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class KeywordsController < ApplicationController
+  before_action(only: %i[new create]) { login_required(t('keyword.page')) }
+  before_action :fetch_file, only: :create
+  before_action :csv?, only: :create
+  before_action :csv_keyword_in_range?, only: :create
+
+  def new
+    render locals: {
+      keyword: Keyword.new
+    }
+  end
+
+  def create
+    Keyword.transaction do
+      CSV.foreach(@file.path, headers: true) do |row|
+        current_user.keywords.create(keyword: row[0])
+      end
+    end
+
+    redirect_to keywords_path, notice: t('keyword.upload_csv_successfully')
+  end
+
+  private
+
+  def fetch_file
+    @file = params[:keyword][:file]
+  end
+
+  def csv?
+    file_type = @file.content_type
+
+    redirect_to keywords_path, alert: t('keyword.file_must_be_csv') unless file_type == 'text/csv'
+  end
+
+  def csv_keyword_in_range?
```

Comment:
`csv_keyword_in_range?` seems like the boolean method, but that doesn't return true/false, it contains the `redirect action` so it makes some confusing there.

how about calling it as validate_csv so it's the `action`

Reply:
Fixed in ca45da01d3c9a2893f137510562a9f31cb4ee6af